### PR TITLE
feat: recursion working in brillig

### PR DESCRIPTION
**Pull request by sirasistant**
# Description

## Problem\*

Adds recursion support by creating entry point artifacts as independent artifacts that jump to the first function called.

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: crates/noirc_evaluator/src/brillig/brillig_ir/artifact.rs [hunk 1]
```diff
@@ -161,6 +155,8 @@ impl BrilligArtifact {
     /// Brillig artifact (self).
     pub(crate) fn link_with(&mut self, func_label: Label, obj: &BrilligArtifact) {
         // Add the unresolved jumps of the linked function to this artifact.
+        println!("Linking with {}", func_label);
+        println!("Current bytecode: {:?}", self.byte_code);
```
Comment (sirasistant) [inline crates/noirc_evaluator/src/brillig/brillig_ir/artifact.rs:159]:
```suggestion
```

**Review by ludamad — DISMISSED** [review]
LGTM

**Review by kevaundray — APPROVED** [review]
This is good, just comment change

File: crates/noirc_evaluator/src/brillig/brillig_ir/artifact.rs [hunk 2]
```diff
@@ -74,13 +74,16 @@ impl BrilligArtifact {
     }
 
     /// Creates an entry point artifact wrapping the bytecode of the function provided.
```
Comment (sirasistant) [inline crates/noirc_evaluator/src/brillig/brillig_ir/artifact.rs:76]:
```suggestion
    /// Creates an entry point artifact that will jump to the function label provided.
```

**Review by kevaundray — APPROVED** [review]

**Review by kevaundray — APPROVED** [review]

**Pull request merged**